lib/WorseReflection/Core/Type/ArrayLiteral.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Phpactor\WorseReflection\Core\Type;
+
+use Phpactor\WorseReflection\Core\Type;
+use Phpactor\WorseReflection\Core\TypeFactory;
+use Phpactor\WorseReflection\TypeUtil;
+
+class ArrayLiteral extends ArrayType implements Literal, Generalizable
+{
+    /**
+     * @var array<array-key,Type>
+     */
+    private array $typeMap;
+
+    /**
+     * @param array<array-key,Type> $typeMap
+     */
+    public function __construct(array $typeMap)
+    {
+        $this->typeMap = $typeMap;
+        $this->keyType = new MixedType();
+        $this->valueType = $this->resolveValueType($typeMap);
+    }
+
+    public function __toString(): string
+    {
+        if (range(0, count($this->typeMap) - 1) === array_keys($this->typeMap)) {
```

Comment:
should probably simplyintroduce a `List` type and resolve this dual responsiblity